app/controllers/topics_controller.rb

```diff
@@ -2,4 +2,19 @@ class TopicsController < ApplicationController
   def index
     @topics = Topic.all
   end
+
+  def new
+    @topic = Topic.new
+  end
+
+  def create
+    Topic.create(topic_params)
+    redirect_to action: "index"
```

Comment:
You can just do ` redirect_to :index`

Reply:
Here the topic may not have actually been created (for example, if it is invalid), so you want to check if the creation succeeds, and, only then, redirect to the index action. If the topic hasn't been created you want to redirect back to the new page for the errors to be fixed.

(hint: You'll also need to instantiate an @topic instance variable)

Reply:
redirect_to :index didn't work, redirect_to topics_path does - this looks better than what I had in the first place.

Reply:
Good catch. My Bad. You can `render :index`, but when you want to redirect, you're quite right, you have to use a route helper (i.e. `some_path(someparam)`)